tools/libs/light/libxl_pcid.c

```diff
@@ -854,6 +854,49 @@ static void server_fini_one(libxl__gc *gc, struct vchan_client *client)
     free(client);
 }
 
+static bool is_vchan_exist(libxl_ctx *ctx, char *watch_dir)
+{
+    GC_INIT(ctx);
+    char *xs_path, *str;
+    char **dir = NULL;
+    unsigned int nb, i;
+    libxl_domid domid;
+    int res;
+    bool ret = false;
+
+
+    pthread_mutex_init(&vchan_client_mutex, NULL);
```

Comment:
It is strange to initialize mutex in such function. 

I believe, you might want to take it there, not init.

Reply:
Yes. It is mistake